demos/multiplatform/task_scheduler/source/main.cpp

```diff
@@ -1,37 +1,39 @@
-#include <FreeRTOS.h>
-#include <semphr.h>
-
-#include <cstdarg>
-
 #include "L3_Application/task.hpp"
+#include "semphr.h"
 #include "utility/log.hpp"
 
 class PrinterTask final : public sjsu::rtos::Task<512>
 {
  public:
-  constexpr PrinterTask(const char * task_name, const char * message)
-      : Task(task_name, sjsu::rtos::Priority::kMedium),
+  PrinterTask(const char * task_name,
+              const char * message,
+              sjsu::rtos::TaskSchedulerInterface & task_scheduler)
+      : Task(task_name, sjsu::rtos::Priority::kMedium, task_scheduler),
         message_(message),
         run_count_(0)
   {
-    print_mutex = { 0 };
-  };
+    print_mutex = NULL;
+  }
+
   bool Setup() override
   {
     if (print_mutex == NULL)
     {
-      print_mutex = xSemaphoreCreateMutex();
+      print_mutex = xSemaphoreCreateMutexStatic(&print_mutex_stack_buffer);
+      // print_mutex = xSemaphoreCreateMutex();
     }
-    LOG_INFO("Setup() for: %s", kName);
+    LOG_INFO("Completed Setup() for: %s", GetName());

```

Comment:
Currently unsure why printing causes a hard fault when using FreeRTOS dynamic allocation.

Reply:
The problem still persists even after removing the extra const.

Reply:
That's strange. May require using a debugger to investigate what is really going on. I highly recommend you pick up a cheap STLink debugger to debug the SJOne or any micro for that matter.